### PR TITLE
Initial Arrowhead Draft

### PR DESCRIPTION
**Pull request by Tornquist**

### CODE REVIEW

File: arrowhead/arrowhead.rb [hunk 1]
```diff
@@ -15,20 +15,43 @@ class Arrowhead
     },
   }
 
+  INVALID_REGION = 0
+  INVALID_SHAPE = 1
+  VALID_REGION_SHAPE = 2
```
Comment (djkotowski) [inline arrowhead/arrowhead.rb]: Avoid using constants as enumerations.  Symbols are the preferred method in Ruby, though in this case, they aren't necessary either.

File: arrowhead/arrowhead.rb [hunk 2]
```diff
@@ -15,20 +15,43 @@ class Arrowhead
...
-    if CLASSIFICATIONS.include? region
-      shapes = CLASSIFICATIONS[region]
-      if shapes.include? shape
-        arrowhead = shapes[shape]
-        "You have a(n) '#{arrowhead}' arrowhead. Probably priceless."
-      else
-        raise "Unknown shape value. Are you sure you know what you're talking about?"
-      end
+    state = check_status(region, shape)
+    case state
```
Comment (djkotowski) [inline arrowhead/arrowhead.rb]: `raise` statements are still valuable in this case.  They help you stop execution when validation errors occur.

[Guard clauses](http://www.thechrisoshow.com/2009/02/16/using-guard-clauses-in-your-ruby-code/) will help you control the execution and greatly shorten the length of your `classify` method here.  No case statement needed.

File: arrowhead/arrowhead.rb [hunk 3]
```diff
@@ -15,20 +15,43 @@ class Arrowhead
...
+    elsif valid_region?(region)
+      INVALID_SHAPE
     else
-      raise "Unknown region, please provide a valid region."
+      INVALID_REGION
     end
   end
+
+  def valid_region?(region)
+    CLASSIFICATIONS.include? region
```
Comment (djkotowski) [inline arrowhead/arrowhead.rb]: Wrap arguments in parentheses.

File: arrowhead/arrowhead.rb [hunk 4]
```diff
@@ -15,20 +15,43 @@ class Arrowhead
...
+  end
+
+  def valid_shape?(region, shape)
+    CLASSIFICATIONS[region].include? shape
```
Comment (djkotowski) [inline arrowhead/arrowhead.rb]: Wrap arguments in parentheses.

File: arrowhead/arrowhead.rb [hunk 5]
```diff
@@ -15,20 +15,43 @@ class Arrowhead
...
+  rescue
```
Comment (djkotowski) [inline arrowhead/arrowhead.rb]: Why should we be rescuing here?  Shouldn't we have caught a missing region before this method was called?

**Comment by djkotowski** [comment]
Good first iteration.  Take a look at the comments and refactor where needed.  Let me know if you have any questions.

File: arrowhead/arrowhead.rb [hunk 6]
```diff
@@ -15,20 +15,43 @@ class Arrowhead
...
-        raise "Unknown shape value. Are you sure you know what you're talking about?"
-      end
+    state = check_status(region, shape)
+    case state
+    when VALID_REGION_SHAPE
+      arrowhead = CLASSIFICATIONS[region][shape]
+      puts "You have a(n) '#{arrowhead}' arrowhead. Probably priceless."
+    when INVALID_SHAPE
+      puts "Unknown shape value. Are you sure you know what you're talking about?"
+    when INVALID_REGION
```
Comment (Tornquist) [inline arrowhead/arrowhead.rb]: Ruby Style Guide:  Indent when as deep as case. I know that many would disagree with this one, but it's the style established in both "The Ruby Programming Language" and "Programming Ruby".

File: arrowhead/arrowhead.rb [hunk 7]
```diff
@@ -15,20 +15,43 @@ class Arrowhead
...
-      shapes = CLASSIFICATIONS[region]
-      if shapes.include? shape
-        arrowhead = shapes[shape]
-        "You have a(n) '#{arrowhead}' arrowhead. Probably priceless."
-      else
-        raise "Unknown shape value. Are you sure you know what you're talking about?"
-      end
+    state = check_status(region, shape)
+    case state
+    when VALID_REGION_SHAPE
```
Comment (Tornquist) [inline arrowhead/arrowhead.rb]: Ruby Style Guide:  Indent when as deep as case. I know that many would disagree with this one, but it's the style established in both "The Ruby Programming Language" and "Programming Ruby".

File: arrowhead/arrowhead.rb [hunk 8]
```diff
@@ -15,20 +15,43 @@ class Arrowhead
...
+      arrowhead = CLASSIFICATIONS[region][shape]
+      puts "You have a(n) '#{arrowhead}' arrowhead. Probably priceless."
+    when INVALID_SHAPE
```
Comment (Tornquist) [inline arrowhead/arrowhead.rb]: Ruby Style Guide:  Indent when as deep as case. I know that many would disagree with this one, but it's the style established in both "The Ruby Programming Language" and "Programming Ruby".

Comment (djkotowski) [inline arrowhead/arrowhead.rb]: It's a change supported by Level UP's [Hound file](https://github.com/jmmastey/level_up_exercises/blob/master/.hound.yml#L40)

Comment (Tornquist) [inline arrowhead/arrowhead.rb]: What is preferred internally though? Making that change makes the final "end" awkward looking. I removed the case statement in my next iteration, but I'd still like to follow best practices.

Comment (djkotowski) [inline arrowhead/arrowhead.rb]: 8b uses the official Ruby style.  I've also seen some projects use this indentation.

**Pull request closed without merging**